.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,29 @@
+name: golangci-lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        working-directory:
+          - ""
+          - test
+          - hack/tools
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+      - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+        with:
+          go-version: 1.19
+      - name: golangci-lint
```

Comment:
We have a `lint` target in ourt Makefile. It may be better to invoke that instead of calling golangci-lint directly in the action. wdyt?

Reply:
Does that use binary?

In the [golang-ci](https://golangci-lint.run/usage/install/#github-actions) doc it's mentioned that it's recommended to use the GitHub action GitHub projects due to some smart caching it implements inside it that makes it fast.

Reply:
If we wanted to use the action then we could use the same approach as upstream CAPI to make sure the the make target (when run locally) uses the same version of golanci-lint: https://github.com/kubernetes-sigs/cluster-api/blob/main/Makefile#L1200:L1203

Reply:
Make sense